uri.go

```diff
@@ -5,59 +5,249 @@ package stun
 
 import (
 	"errors"
-	"fmt"
+	"net"
 	"net/url"
 	"strconv"
 )
 
-// Scheme definitions from RFC 7064 Section 3.2.
+var (
+	// ErrUnknownType indicates an error with Unknown info.
+	ErrUnknownType = errors.New("Unknown")
+
+	// ErrSchemeType indicates the scheme type could not be parsed.
+	ErrSchemeType = errors.New("unknown scheme type")
+
+	// ErrSTUNQuery indicates query arguments are provided in a STUN URL.
+	ErrSTUNQuery = errors.New("queries not supported in stun address")
+
+	// ErrInvalidQuery indicates an malformed query is provided.
+	ErrInvalidQuery = errors.New("invalid query")
+
+	// ErrHost indicates malformed hostname is provided.
+	ErrHost = errors.New("invalid hostname")
+
+	// ErrPort indicates malformed port is provided.
+	ErrPort = errors.New("invalid port")
+
+	// ErrProtoType indicates an unsupported transport type was provided.
+	ErrProtoType = errors.New("invalid transport protocol type")
+)
+
+// SchemeType indicates the type of server used in the ice.URL structure.
+type SchemeType int
+
+// Unknown defines default public constant to use for "enum" like struct
+// comparisons when no value was defined.
+const Unknown = iota
```

Comment:
Do you think it reads cleaner type wise to just define SchemeTypeUnknown and ProtoTypeUnknown?

Reply:
Agreed.

I've originally thought to keep the code 1:1 as in `pion/ice` from where it originates.
The motivation behind this was to avoid breaking the API in `pion/ice`, by simply adding a type alias for `struct URI`. But this type alias is still possible and we can still define the `Unknown` constant in `pion/ice` to maintain API compatability.

Reply:
sgtm